kratos/modeler/clean_up_problematic_triangles_modeler.cpp

```diff
@@ -0,0 +1,306 @@
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/timer.h"
+#include "utilities/entities_utilities.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "modeler/clean_up_problematic_triangles_modeler.h"
+
+namespace Kratos
+{
+
+void CleanUpProblematicTrianglesModeler::SetupModelPart()
+{
+    KRATOS_TRY;
+
+    Timer::Start("CleanUpProblematicTrianglesModeler");
+
+    // We clean the mesh
+    const std::string entity_type = mParameters["entity_type"].GetString();
+    const IndexType first_node_id = mParameters["first_node_id"].GetInt();
+    const IndexType first_element_id = mParameters["first_element_id"].GetInt();
+    const IndexType first_condition_id = mParameters["first_condition_id"].GetInt();
+    const double area_tolerance = mParameters["area_tolerance"].GetDouble();
+    CleanUpProblematicGeometriesInMesh(mrModelPart, entity_type, first_node_id, first_element_id, first_condition_id, area_tolerance);
+
+    Timer::Stop("CleanUpProblematicTrianglesModeler");
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void CleanUpProblematicTrianglesModeler::CleanUpProblematicGeometriesInMesh(
+    ModelPart& rThisModelPart,
+    const std::string& rEntityType,
+    const IndexType FirstNodeId,
+    const IndexType FirstElementId,
+    const IndexType FirstConditionId,
+    const double AreaTolerance
+    )
+{
+    if (rEntityType == "element") {
+        CleanUpProblematicGeometries<Element>(rThisModelPart, FirstNodeId, FirstElementId, AreaTolerance);
+    } else if (rEntityType == "condition") {
+        CleanUpProblematicGeometries<Condition>(rThisModelPart, FirstNodeId, FirstConditionId, AreaTolerance);
+    } else if (rEntityType == "geometry") { // TODO: Implement geometry cleanup
+        KRATOS_ERROR << "Geometry cleanup not yet supported. Sorry" << std::endl;
+    } else {
+        KRATOS_ERROR << "Unknown entity type: " << rEntityType << std::endl;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <typename TEntityType>
+void CleanUpProblematicTrianglesModeler::CleanUpProblematicGeometries(
+    ModelPart& rThisModelPart,
+    const IndexType FirstNodeId,
+    const IndexType FirstEntityId,
+    const double AreaTolerance
+    )
+{
+    // Get the entities container
+    auto& r_entities = EntitiesUtilities::GetEntities<TEntityType>(rThisModelPart);
+
+    // Initialize variables
+    double average_length = 0.0;
+    double ref_area = 0.0;
+    {
```

Comment:
```suggestion
```
I think you left this from the previous parallel block.

Reply:
Nope, it is to automatically destry variable sonce out of scope